src/parse_utils.rs

```diff
@@ -170,3 +176,64 @@ pub(crate) fn consume_comma(tokens: &mut TokenIter) -> Option<Punct> {
         _ => None,
     }
 }
+
+/// Parse `::`, as in path separator or turbofish.
+///
+/// Does not advance `tokens` if the double colon is not found.
+pub(crate) fn try_consume_colon2(tokens: &mut TokenIter) -> Option<[Punct; 2]> {
+    let before = tokens.clone();
+
+    if let Some(TokenTree::Punct(first)) = tokens.next() {
+        if first.as_char() == ':' && first.spacing() == Spacing::Joint {
+            if let Some(TokenTree::Punct(second)) = tokens.next() {
+                if second.as_char() == ':' && second.spacing() == Spacing::Alone {
+                    return Some([first, second]);
+                }
+            }
+        }
+    }
+
+    *tokens = before;
```

Comment:
I'm not crazy about this part.

I get why it's necessary, but it has the potential to clone a lot of token, which is annoying given that the whole point of venial is to give good performance.

On the long term, I'd rather have a token iterator with more lookahead for that kind of situation, though for now we'll probably merge this as-is.

Reply:
Added `TODO`